### ISSUE TITLE
CLI generators not so random

### ISSUE THREAD

**Issue by mindjuice** [victorquinn/chancejs#142]
When running generators from the new CLI feature, they are not very random.  Here is the output of calling "chance word" several times in a row:

vapik
teogefe
suvaflud
ut
gibgewuga
cim
gendadel
ate
rizjefjo
ro
cim
cof
cim
mi
gibgewuga
gibgewuga
gendadel
ut
teogefe
alogikfur

You can see "gibgewuga" is repeated 3 times, as is "cim".  "gendadel" and "teogefe" are repeated twice.

It looks like this is not specific to "word", but rather systemic.  Similar results occur with "chance name":

Madge Scott
Flora Vaughn
Elsie Wallace
Victoria Huff
Ricky Russell
Rodney Thornton
Bobby Swanson
Ida Carlson
Dennis Atkins
Emily Bass
Mina Sims
Bettie Quinn
James Hamilton
Esther Young
Anne Weaver
Rodney Thornton
Anne Weaver
Bernice Dunn
Lura Guzman
Curtis Norris
Jesus Kim
Jesus Kim
Bernice Dunn
Bernice Dunn
Francis Ward
Bernice Dunn
Emily Bass
Esther Young
Bettie Quinn

I haven't investigated yet, but I assume this is related to not having a true source of entropy at startup.  For the CLI to be useful, I think it needs a better source of entropy.  I'm sure even the CPU tick time would give better results.

**Comment by victorquinn** [comment]
Hmm, odd, will investigate, that certainly looks problematic, particularly the name generation!

Internally it is using a hand-rolled Mersenne Twister (with the current hammertime as the seed) so it shouldn't generate such crappy pseudo-random results when no seed is provided. There are a bunch of tests which verify this so something seems awry here.

That said, we could use CPU tick time, though that may not always be available. Chance was originally conceived as a way to generate random things in the browser where such low level access is not available. Node.js came next and the CLI use case is relatively new. 

Now that 2/3 of the use cases are non-browser, it may make sense to revisit that and/or make it use a better random source when in a non-browser environment (as in #77 ) though I want to ensure any such solution does not break any browser use (and #77 has been sitting there unmerged as it has failing tests).

**Comment by victorquinn** [comment]
I should add I have been able to reproduce, calling `chance word` in the cli only 9 times presented a dupe which should not be happening! So there is 100% some issue here, will investigate!

**Comment by mindjuice** [comment]
I was originally testing inside a Vagrant Ubuntu VM, and started to wonder if there was some issue with that, but I was also able to reproduce on my Mac running Yosemite.

**Comment by oliversalzburg** [comment]
I had also already noticed this when I implemented the CLI.

One of the original problems was fixed in https://github.com/victorquinn/chancejs/commit/c66d18b1b840170f4047dac88653e536379c0001. The code that produces the "real" seed from the original seed seems a bit problematic to me. This is where I would start looking.

**Comment by jokeyrhyme** [comment]
Maybe this is related to:
- https://medium.com/@betable/tifu-by-using-math-random-f1c308c4fd9d#.3yja09hzj
- https://code.google.com/p/v8/issues/detail?id=4566

**Comment by victorquinn** [comment]
@jokeyrhyme nope, not related. We use a Mersenne Twister under the hood and there is zero reliance on Math.random.

I am fairly certain this issue was resolved a long time ago, this issue was just never closed but I'll verify and close if so.

**Comment by bvaughn** [comment]
FWIW I just ran a quick test myself, generating 1,000 random words, and noted 31 words repeating twice, 11 repeating 3 times, 6 repeating 4 times, and 3 repeating 5 times.

**Comment by JavaScriptBach** [comment]
+1 to this issue. `chance.word()` has poor randomness because it defaults to 1-3 syllables. The workaround we use is to call e.g. `chance.word({length: 20})`.

I think the default for this function and all chance functions should be changed to generate a high degree of randomness. Maybe it would require publishing a new major version, but I think it's worth getting the default right. It's surprising behavior for a user to call chance.word() a few times and get a collision.

**Comment by victorquinn** [comment]
The way `chance.word()` was intended to be used was not necessarily to create an entirely unique and random word that would not have a collision.

It was to create something that generally looked like a real word that could be used for generating longer bodies of text that would generally look like real-world text with functions like `chance.paragraph()` (think lorem ipsum)

So it's not terribly surprising that you get a collision. If you want something that is far less likely to have a collision you could use `chance.guid()` or yes you could generate a word with a longer length. 

But this generally is the expected behavior here for `chance.word()` and doesn't necessarily mean it has poor randomness, just that it will generate some short words that are likely to have collisions
